x/stake/keeper.go

```diff
@@ -298,3 +298,33 @@ func (k Keeper) setParams(ctx sdk.Context, params Params) {
 	store.Set(ParamKey, b)
 	k.params = Params{} // clear the cache
 }
+
+//_______________________________________________________________________
+
+// load/save the pool
+func (k Keeper) GetPool(ctx sdk.Context) (gs Pool) {
+	// check if cached before anything
+	if k.gs != (Pool{}) {
```

Comment:
Is a layer of caching on top of the multistore necessary?

Reply:
this is based off of conversation with jae/sunny in berlin... if retrieving multiple times why unmarshal multiple times